assets/js/components/user-input/UserInputPreview.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * User Input Preview.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useCallback, useState, Fragment } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { STORE_NAME as CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import { STORE_NAME as CORE_SITE } from '../../googlesitekit/datastore/site/constants';
+import { Cell, Row } from '../../material-components';
+import Button from '../button';
+import ProgressBar from '../ProgressBar';
+import ErrorNotice from '../ErrorNotice';
+import UserInputPreviewGroup from './UserInputPreviewGroup';
+import UserInputQuestionNotice from './UserInputQuestionNotice';
+import { getUserInputAnwsers } from './util/constants';
+const { useSelect, useDispatch } = Data;
+
+export default function UserInputPreview( { back, goTo } ) {
+	const [ isNavigating, setIsNavigating ] = useState( false );
+
+	const dashboardURL = useSelect( ( select ) => select( CORE_SITE ).getAdminURL( 'googlesitekit-dashboard' ) );
+	const settings = useSelect( ( select ) => select( CORE_USER ).getUserInputSettings() );
+	const { isSavingSettings, error } = useSelect( ( select ) => ( {
+		isSavingSettings: select( CORE_USER ).isFetchingSaveUserInputSettings(),
+		error: select( CORE_USER ).getErrorForAction( 'saveUserInputSettings', [] ),
+	} ) );
+
+	const { saveUserInputSettings } = useDispatch( CORE_USER );
+	const submitChanges = useCallback( async () => {
+		setIsNavigating( true );
+		const response = await saveUserInputSettings();
+		if ( ! response.error ) {
+			global.location.assign( dashboardURL );
+		} else {
+			setIsNavigating( false );
+		}
+	}, [ dashboardURL ] );
+
+	const {
+		USER_INPUT_ANSWERS_GOALS,
+		USER_INPUT_ANSWERS_HELP_NEEDED,
+		USER_INPUT_ANSWERS_POST_FREQUENCY,
+		USER_INPUT_ANSWERS_ROLE,
+	} = getUserInputAnwsers();
+
+	return (
+		<div className="googlesitekit-user-input__preview">
+			<Row>
+				<Cell lgSize={ 12 } mdSize={ 8 } smSize={ 4 }>
+					{ ( isSavingSettings || isNavigating ) && ! error && (
+						<ProgressBar />
+					) }
+					{ ( ( ! isSavingSettings && ! isNavigating ) || error ) && (
```

Comment:
I don't think the `error` check is needed in either of these. If there is an error, per the behavior both `isSavingSettings` and `isNavigating` will always be `false`. So I think we can omit that.

Reply:
Yes, you are right. Removed it.